konduit-serving-core/src/main/java/ai/konduit/serving/util/image/NativeImageLoader.java

```diff
@@ -176,13 +177,11 @@ static Mat convert(PIX pix) {
                     pix2 = pixConvert4To8(pix, 0);
                     break;
                 case 8:
-                    pix2 = pix;
-                    break;
                 case 24:
                     pix2 = pix;
                     break;
                 default:
-                    assert false;
+                    Preconditions.checkState(false, String.format("Unrecognized pixel depth of %d", pix.d()));
```

Comment:
Better:
`throw new IllegalStateException("Unrecognized pixel depth of " + pix.d()");`
or
`Preconditions.throwStateEx("Unrecognized pixel depth of %s", pix.d());`